File: gm-utils/js/src/commom/utils/string/StringExtraiNumeros.js
import ArrayLst from '../array/ArrayLst';
import StringEmpty from './StringEmpty';
import StringExtraiCaracteres from './StringExtraiCaracteres';

export default class StringExtraiNumeros {

	static exec(s) {
		return StringExtraiCaracteres.exec(s, StringExtraiNumeros.NUMEROS);
	}

	static nullIfEmpty(s) {
		s = StringExtraiNumeros.exec(s);
		if (StringEmpty.is(s)) {
			return null;
		}
		return s;
	}

}
StringExtraiNumeros.NUMEROS = ArrayLst.build("0", "1", "2", "3", "4", "5", "6", "7", "8", "9");

File: gm-utils/src/main/java/src/commom/utils/string/StringExtraiNumeros.java
package src.commom.utils.string;

public class StringExtraiNumeros {

	private StringExtraiNumeros() {}
	
	private static final String NUMEROS = "0123456789";

	public static String exec(String s) {
		return StringExtraiCaracteresBaseadoEmString.exec(s, NUMEROS);
	}

	public static String nullIfEmpty(String s) {
		return StringExtraiCaracteresBaseadoEmString.nullIfEmpty(s, NUMEROS);
	}
	
}
